app/src/main/java/org/dharmaseed/android/TalkFetcherTask.java

```diff
@@ -48,4 +59,31 @@ protected Void doInBackground(Void... params) {
         publishProgress();
         return null;
     }
+
+    @Override
+    protected void extraTableProcessing(DBManager dbManager, JSONObject talks) throws JSONException {
+        Iterator<String> it = talks.keys();
+        while (it.hasNext()) {
+            JSONObject talk = talks.getJSONObject(it.next());
+            int talkID = talk.getInt("id");
+
+            // First, clear any existing entries for this talk, so that if any teachers are ever
+            // removed from a talk after the talk is already added, we'll pick up that change
+            // in the database.
+            dbManager.deleteID(talkID, DBManager.C.TalkTeachers.TABLE_NAME);
```

Comment:
Haha yeah, if at all possible I wanted to avoid writing a new database synchronization bug, especially after all of your great work to fix the last ones! :joy: